rotkehlchen/accounting/structures/base.py

```diff
@@ -119,22 +119,22 @@ def __init__(
         self.notes = notes
         self.identifier = identifier
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: object) -> bool:
         if type(self) is not type(other):  # pylint: disable=unidiomatic-typecheck
             return False
```

Comment:
why not to use the isinstance as suggested here? https://pylint.pycqa.org/en/latest/user_guide/messages/convention/unidiomatic-typecheck.html

Reply:
Because it's not the same thing. Perhaps we should add a comment there.

let's say you have a `Parent` class and a `Child` class inheriting from `Parent`. Try this:


```python
class Parent:

    def __init__(self, x: int) -> None:
        self.x = 5

    def __eq__(self, other: object):
        if not isinstance(other, Parent):
        # if type(other) != type(self):
            return False

        return self.x == other.x
            

class Child(Parent):
    pass


class OtherChild(Parent):
    pass


child = Child(x=42)
otherchild = OtherChild(x=42)

print(child == otherchild)  # prints True for isinstance and False (what you want) for type check
```